### PR TITLE
DAOS-5982 control: Pool create on available ranks

### PR DESCRIPTION
**Pull request by mjmac**
Pool create requests should only include available ranks
in order to avoid failures when trying to contact down ranks.

Also fixes a goroutine counting bug in a test.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]

File: src/control/system/database_test.go [hunk 1]
```diff
@@ -74,6 +74,52 @@ func waitForLeadership(t *testing.T, ctx context.Context, db *Database, gained b
...
+	for i, ms := range memberStates {
+		if err := db.AddMember(MockMember(t, uint32(i), ms)); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	for name, tf := range map[string]func(t *testing.T){
+		"individual state filters": func(t *testing.T) {
+			for _, ms := range memberStates {
+				matchLen := len(db.filterMembers(ms))
```
Comment (kjacque) [inline src/control/system/database_test.go:97]: Don't we also want to be sure that it's returning the correct members, not just the right number?

File: src/control/system/member.go [hunk 2]
```diff
@@ -45,41 +45,57 @@ type MemberState int
 
 const (
 	// MemberStateUnknown is the default invalid state.
```
Comment (jbrosenz) [inline src/control/system/member.go:47]: If it doesn't violate a style rule, I would have preferred to see these values in hex, given that they are used as bit masks.  Just seems more natural.

**Review by jbrosenz — APPROVED** [review]
Looks fine to me.  I just had one comment which does not affect functionality.

File: src/control/system/member.go (hunk 2 above)
Comment (mjmac) [reply src/control/system/member.go:47]: Done.

File: src/control/system/database_test.go (hunk 1 above)
Comment (mjmac) [reply src/control/system/database_test.go]: Good call, fixed.

**Review by jbrosenz — DISMISSED** [review]

**Review by tanabarr — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by mjmac** [comment]
Known failures:
   - [DAOS-5992](https://jira.hpdd.intel.com/browse/DAOS-5992)
   - [DAOS-6025](https://jira.hpdd.intel.com/browse/DAOS-6025)

**Pull request merged**